runtime-modules/content-directory/src/lib.rs

```diff
@@ -0,0 +1,1645 @@
+            actor: Actor<T>,
+            entity_id: T::EntityId,
+            in_class_schema_property_id: PropertyId,
+            index_in_property_vec: VecMaxLength,
+            property_value: SinglePropertyValue<T>,
+            nonce: T::Nonce
+        ) -> dispatch::Result {
+            let account_id = ensure_signed(origin)?;
+
+            Self::ensure_known_entity_id(entity_id)?;
+
+            let (entity, access_level) = Self::get_entity_and_access_level(account_id, entity_id, actor)?;
+
+            // Try to find a current property value in the entity
+            // by matching its id to the id of a property with an updated value.
+            if let Some(PropertyValue::Vector(entity_prop_value)) =
+                entity.values.get(&in_class_schema_property_id)
+            {
+                let class_prop = Self::ensure_class_property_type_unlocked_for(
+                    entity.class_id,
+                    in_class_schema_property_id,
+                    access_level,
+                )?;
+
+                // Ensure property value vector nonces equality to avoid possible data races,
+                // when performing vector specific operations
+                entity_prop_value.ensure_nonce_equality(nonce)?;
+
+                // Validate a new property value against the type of this property
+                // and check any additional constraints like the length of a vector
+                // if it's a vector property or the length of a text if it's a text property.
+                class_prop.ensure_prop_value_can_be_inserted_at_prop_vec(
+                    &property_value,
+                    entity_prop_value,
+                    index_in_property_vec,
+                    entity.get_permissions().get_controller(),
+                )?;
+            };
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Insert property value into property value vector
+            <EntityById<T>>::mutate(entity_id, |entity| {
+                let value = property_value.get_value();
+                if let Some(entities_rc_to_increment) = value.get_involved_entity() {
+                    Self::increment_entities_rc(&[entities_rc_to_increment]);
+                }
+                if let Some(PropertyValue::Vector(current_prop_value)) =
+                    entity.values.get_mut(&in_class_schema_property_id)
+                {
+                    current_prop_value.vec_insert_at(index_in_property_vec, value)
+                }
+            });
+
+            Ok(())
+        }
+
+        pub fn transaction(origin, actor: Actor<T>, operations: Vec<OperationType<T>>) -> dispatch::Result {
+            // This map holds the T::EntityId of the entity created as a result of executing a CreateEntity Operation
+            // keyed by the indexed of the operation, in the operations vector.
+            let mut entity_created_in_operation: BTreeMap<usize, T::EntityId> = BTreeMap::new();
```

Comment:
I know this is old code, but should this really be a map? isnt it really a vector?

Reply:
Good point, i did not refactored it much. Switched to `Vec` here https://github.com/iorveth/substrate-runtime-joystream/commit/9d0312aadfc961b1a54a84f5606c6a6d80aa59fe